supervisor/shared/board.c

```diff
@@ -44,3 +44,11 @@ MP_WEAK void board_deinit(void) {
 // Do-nothing so not all boards need to provide this function.
 MP_WEAK void reset_board(void) {
 }
+
+// Do-nothing so not all boards need to provide this function.
+MP_WEAK void mp_board_init(void) {
+}
+
+// Do-nothing so not all boards need to provide this function.
+MP_WEAK void board_gc_collect(void) {
+}
```

Comment:
Do you actually need these? I'd love to move away from MP_WEAK since these can be used even when not expected. In other words, the compiler doesn't check for correctness here. A #define to enable the function would be compile checked.

Reply:
> Did you mean that one might accidentally replace a weak function by mistake?

Nope, you may *think* you are overriding one but don't actually. So, it'll compile and link but not run the implementation you expected. (Lots of IDF issues for us were this.)